packages/aws-cdk-lib/aws-ecs/lib/base/service-managed-volume.ts

```diff
@@ -0,0 +1,190 @@
+import { PropagatedTagSource } from './base-service';
+import * as ec2 from '../../../aws-ec2';
+import * as iam from '../../../aws-iam';
+import * as kms from '../../../aws-kms';
+import { BaseMountPoint, ContainerDefinition } from '../container-definition';
+
+/**
+* Represents the Volume configuration for an ECS service.
+*/
+export interface ServiceVolumeProps {
+  /**
+  * The name of the volume. This corresponds to the name provided in the ECS TaskDefinition.
+  */
+  readonly name: string;
+
+  /**
+  * Configuration for an Amazon Elastic Block Store (EBS) volume managed by ECS.
+  *
+  * @default - undefined
+  */
+  readonly managedEBSVolume?: ServiceManagedEBSVolumeConfiguration
+}
+
+/**
+* Represents the configuration for an ECS Service managed EBS volume.
+*/
+export interface ServiceManagedEBSVolumeConfiguration {
+  /**
+  * An IAM role that allows ECS to make calls to EBS APIs on your behalf.
+  * This role is required to create and manage the Amazon EBS volume.
+  */
+  readonly role: iam.IRole;
```

Comment:
I think it would be user friendly to make this optional and create it internally if unspecified, what do you think?

```ts
    // example
    this.role = props.role ?? new iam.Role(this, 'Role', {
      assumedBy: new iam.ServicePrincipal('ecs.amazonaws.com'),
      managedPolicies: [
        iam.ManagedPolicy.fromAwsManagedPolicyName('service-role/AmazonECSInfrastructureRolePolicyForVolumes'),
      ],
    });
```

Reply:
That's a pretty good idea 👍

Reply:
made that role optional and a default role over here [[link]](https://github.com/aws/aws-cdk/pull/28691/files#diff-c9cc7970a22bbdf1add32b7adcb50af5f8bc6212db0340715d335b9446ef31e9R219-R224)